src/Compilers/CSharp/Portable/Binder/Binder_Lambda.cs

```diff
@@ -94,6 +94,14 @@ internal partial class Binder
 
                 foreach (var p in parameterSyntaxList.Value)
                 {
+                    if (p.AttributeLists.Count > 0 && syntax.Kind() == SyntaxKind.AnonymousMethodExpression)
```

Comment:
It was literally just to keep behavior the same as today :)   If you want, i'm happy to change the conditional to be broader.